x-pack/plugins/ml/server/models/job_service/new_job_caps/field_service.ts

```diff
@@ -90,8 +90,8 @@ class FieldsService {
             fields.push({
               id: k,
               name: k,
-              type: field.type,
-              aggregatable: field.aggregatable,
+              type: field.type as ES_FIELD_TYPES,
+              aggregatable: field.time_series_metric !== 'counter' ?? field.aggregatable,
```

Comment:
Would be good to move this check into a function for easier maintenance since the logic might change later on, and I suspect it might not be just AD that might be affected.

 Would also be good to add a comment to reference why we are not currently supporting time_series_metric as an aggregatabe field.

Reply:
Updated in https://github.com/elastic/kibana/pull/153021/commits/d28c820572dc13077cdd0eb2c6da1644668ee0c1